crates/tako/src/internal/server/reactor.rs

```diff
@@ -162,32 +162,22 @@ pub(crate) fn on_remove_worker(
 
 pub(crate) fn on_new_tasks(core: &mut Core, comm: &mut impl Comm, new_tasks: Vec<Task>) {
     assert!(!new_tasks.is_empty());
-
-    let mut task_map: Map<_, _> = new_tasks.into_iter().map(|t| (t.id, t)).collect();
-    let ids: Vec<_> = task_map.keys().copied().collect();
-    for task_id in ids {
-        let mut task = task_map.remove(&task_id).unwrap();
-
+    for mut task in new_tasks.into_iter() {
```

Comment:
We should document the assumption that there is a topological ordering. Maybe also a `debug_assert` or a check gated behing `#[cfg(debug_assertions)]`?

Reply:
`validate_submit` will now return `InvalidDependencies` if graph is not in topological order.